src/models/model.ts

```diff
@@ -153,3 +141,15 @@ export function executeModel(props: {
     },
   });
 }
+
+function objectToURLSearchParams(object: Record<string, unknown>) {
+  const params = new URLSearchParams();
+  for (const key in object) {
+    if (typeof object[key] === 'object') {
```

Comment:
Minor – Let's use `isPureObject` from utils, instead of the typeof check here. Probably doesn't matter, but this way "null" falls through to the next case and we can do a null or undefined check.

Reply:
And will the `isPureObject` check catch arrays too?
Right now, checking for `object` types we are stringifying object and arrays, and converting `null` to `"null"`.

However, we could separate these three cases in three conditional to make it easier to read

Reply:
It would read as something like this
```
if (isPureObject(object[key])) {
  params.append(key, JSON.stringify(object[key]));
}
else if (Array.isArray(object[key])) {
  params.append(key, JSON.stringify(object[key]));
}
else if (object[key] === null) {
  params.append(key, 'null');
}
else if (object[key] !== undefined) {
  params.append(key, String(object[key]));
}
```

Reply:
Something Im not sure about is how we want to treat `null` values here